jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js

```diff
@@ -53,18 +57,44 @@ YUI.add('machine-view-add-machine', function() {
     /**
       Create the machine.
 
-      @method _createMachine
+      @method _submitForm
     */
-    _createMachine: function() {
-      // Don't try and create a container if the container type has not been
-      // selected.
-      if (this.props.parentId && !this.state.containerType) {
+    _submitForm: function() {
+      var props = this.props;
+      var state = this.state;
+      var machineId = this._getParentId();
+      var selectedContainer = state.selectedContainer;
+      var selectedMachine = state.selectedMachine;
+      var constraints = state.constraints;
+      var machine;
+      if (machineId && machineId !== 'new' && !selectedContainer) {
+        // Don't try and create a container if the container type has not been
+        // selected.
         return;
       }
-      this.props.createMachine(
-        this.state.containerType, this.props.parentId, this.state.constraints);
+      if (selectedMachine === 'new' || selectedContainer === 'lxc' ||
```

Comment:
We probably won't remember why these conditionals are here (and the `props.unit` one below) Could you add comments to the rational here. Thankx